### ISSUE TITLE
Scikit learn vs software engineering

### ISSUE THREAD

**Issue by winchell-jeff** [ldulcic/text-segmentation#2]
You might have good code, but it needs to be rewritten because you relied on an unreliable version 0.x product (scikit learn).

The sample notebook blows up repeatedly over issues because scikit learn doesn't care about code longevity and changed their APIs.

**Comment by ldulcic** [comment]
@winchell-jeff I agree, I wrote this a long time ago, I didn't even generate `requirements.txt`. PRs are very welcome if you would like to help 😊

**Comment by winchell-jeff** [comment]
I like to use mini conda to setup isolated environments and install the code and then have conda generate a yml file. I will send something soon about this for a PR. But I don't know about variations based on the OS.
________________________________

**Comment by ldulcic** [comment]
@winchell-jeff I would appreciate it very much if you would send a PR :)

**Comment by winchell-jeff** [comment]
This will take me more than a few minutes to figure out what needs to be changed in the code for more recent sklearn libraries and provide install version info. I can do this during work hours for me if I finally get a version that works. (With all the other text segmenting libraries I have failed to get the basic hello world/demo app working)

**Comment by ldulcic** [comment]
@winchell-jeff great! Feel free to create a draft PR if you start working on this so I can help you out :)

**Comment by winchell-jeff** [comment]
Sorry, I had zero time coding yesterday afternoon, dealing with admin and bureaucracy. I hope I can do this today.
